File: easy_1/anagrams/ruby/v2/anagram.rb
# anagrams.rb

class Anagram
  def initialize(input_word)
    @key_word = input_word
  end

  def match(word_list)
    sorted_key_word = key_word.downcase.split('').sort
    word_list.select do |word|
      not_key_word = word.downcase != key_word.downcase
      is_an_anagram = word.downcase.split('').sort == sorted_key_word
      not_key_word && is_an_anagram
    end
  end

  private
  attr_reader :key_word
end

File: easy_1/anagrams/js/v1/anagram.js
class Anagram {
  constructor(keyWord) {
    this.keyWord = keyWord;
  }

  match(wordList) {
    const lowerCasedKeyWord = this.keyWord.toLowerCase();
    const sortedKeyword = lowerCasedKeyWord.split('').sort().join();

    return wordList.filter( (word) => {
      const lowerCasedWord = word.toLowerCase();
      const isNotWord = lowerCasedWord !== this.keyWord;
      const isAnagram = sortedKeyword === lowerCasedWord.split('').sort().join();
      return isAnagram && isNotWord;
    });
  }
}

export default Anagram;